lib/communicator_stack.cpp

```diff
@@ -49,8 +50,16 @@ namespace quda
   void push_communicator(const CommKey &split_key)
   {
     if (comm_nvshmem_enabled())
-      errorQuda(
-        "Split-grid is currently not supported with NVSHMEM. Please set QUDA_ENABLE_NVSHMEM=0 to disable NVSHMEM.");
+      errorQuda("Split-grid is currently not supported with NVSHMEM. Set QUDA_ENABLE_NVSHMEM=0 to disable NVSHMEM.");
+
+    // if reverting to global, we will need to join the tunecaches
+    bool join_tune_cache = split_key == default_comm_key;
```

Comment:
Shouldn't we theoretically need to do this each time we go to a `split_key` with a smaller `[key].product()`?

Reply:
Sorry, not following.  What do you mean?

Reply:
If there are recursive split-grid calls, i.e., further splitting an already split grid.

Reply:
Ah, gotcha now.  That is precluded and recursive split grids are not supported.

Reply:
The way to do it would be with a `std::stack` I guess.